### DISCUSSION TITLE
Problem with set_on_audio_state_changed and set_on_volumechange

### DISCUSSION THREAD

**Discussion by thanhtung386** [pschatzmann/ESP32-A2DP#401] [Q&A] [answered]
This is my code:
#include "BluetoothA2DPSink.h"
BluetoothA2DPSink a2dp_sink;
// for esp_a2d_connection_state_t see https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/bluetooth/esp_a2dp.html#_CPPv426esp_a2d_connection_state_t
void connection_state_changed(esp_a2d_connection_state_t state, void *ptr){
  Serial.println(a2dp_sink.to_str(state));
}
// for esp_a2d_audio_state_t see https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/bluetooth/esp_a2dp.html#_CPPv421esp_a2d_audio_state_t
void audio_state_changed(esp_a2d_audio_state_t state, void *ptr){
  Serial.println(a2dp_sink.to_str(state));
}

void volumeChanged(int new_volume) {
Serial.println(new_volume);
}

void setup() {
  Serial.begin(115200);
  a2dp_sink.set_on_connection_state_changed(connection_state_changed);
  a2dp_sink.set_on_audio_state_changed_post(audio_state_changed);
  a2dp_sink.set_on_volumechange(volumeChanged);    
  a2dp_sink.start("MyMusic");  
}
void loop() {
  delay(1000); // do nothing
}

When audio state is "Suspended". After I changed the volume, audio state is automatically change to "Started". Then it changes to "Suspended". This problem doesn't happen when audio state is "Started"
Can you help me solve that problem?
Thanks for your reading!!!

**Comment by pschatzmann** [comment, marked as answer]
I don't see any problem here. 
I guess the volume can only be changed in started state...
